From 8eb3ec51a58dd02334118d8f0aa1bebb5704d15c Mon Sep 17 00:00:00 2001
From: Father Chrysostomos via RT <bug-Tk@rt.cpan.org>
Date: Mon, 6 Aug 2012 17:42:14 -0400
Subject: [PATCH] Fix compilation under threaded bleadperl

 Ticket <URL: https://rt.cpan.org/Ticket/Display.html?id=78824 >

CopSTASHPV is no longer an lvalue under threads.  (It never was under non-threaded builds.)
---
 tkGlue.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/tkGlue.c b/tkGlue.c
index 1b7c025..9079302 100644
--- a/tkGlue.c
+++ b/tkGlue.c
@@ -4154,11 +4154,7 @@ int type;
    SV *x = NULL;
    int prefix = '?';
    name = SvPV(sv,na);
-#ifdef USE_ITHREADS
-   CopSTASHPV(PL_curcop) = NULL;
-#else
-   CopSTASH(PL_curcop) = NULL;
-#endif
+   CopSTASH_set(PL_curcop, NULL);
    switch (type)
     {
      case TK_CONFIG_SCALARVAR:
-- 
1.8.2.3

